theano/compile/ops.py

```diff
@@ -623,21 +604,16 @@ class Rebroadcast(gof.Op):
     c_code_and_version = {}
 
     check_input = False
+    __props__ = ("axis",)
 
     def __init__(self, *axis):
-        self.axis = dict(axis)
+        # Sort them to make sure we merge all possible case.
+        items = sorted(iteritems(axis))
```

Comment:
Actually, axis is a list of pairs, so you cannot call iteritems on it.
I think that should be:
```python
items = sorted(axis)
self.axis = OrderedDict(items)
```
but please try it first.